x/datapool/keeper/pool.go

```diff
@@ -175,9 +176,21 @@ func (k Keeper) CreatePool(ctx sdk.Context, curator sdk.AccAddress, poolParams t
 	// store pool
 	k.SetPool(ctx, newPool)
 
+	// Initialize shareToken supply
+	k.setInitialSupply(ctx, poolID)
```

Comment:
When creating pool, we may be able to set total supply with `poolParams.targetNumData`, and send data share token to seller when selling data.

Reply:
Yes, that is also possible.
If that happens, we will have to mint the shareToken as much as `poolParams.targetNumData` when creating the pool.

It's been a bit of a discussion, if we decide to distribute a reward every time an NFT is sold, we need to exclude for the ShareToken owned by the pool.
If this decision is not accepted, it seems better to mint in advance, as @hansol-medi suggested.

Reply:
If we distribute the rewards from NFT sales to seller and curator before reaching `poolParams.targeNumData` (we need further discussion as you said), we may have to calculate the portion of seller to total number of seller, which is equivalent to `poolParams.targeNumData`. 

In other words, I meant that **mint and send ShareToken every seller sell data** or **mint all ShareToken when creating pool and send every seller sell data**.